_overviews/scala3-book/concurrency.md

```diff
@@ -180,7 +181,7 @@ See the [Futures and Promises][futures] page for a discussion of additional meth
 
 ## Running multiple futures and joining their results
 
-To run multiple futures in parallel and join their results when all of the futures complete, use a `for` expression.
+To run multiple computations in parallel and join their results when all of the futures have been completed, use a `for` expression.
```

Comment:
This sentence was really confusing me. It sounds like it is suggesting:

```
for {
  a <- future { ... }
  b <- future { ... }
  c <- future { ... }
} yield ...
```
but IIUC this would actually run them in sequence, not in parallel, no?

Reply:
You are right, there is a subtlety here, maybe we should add a note about that (the fact that running the computations must be performed _outside_ of the `for` expression to possibly get parallelism)?

Reply:
Consider describing it more clearly and writing a correct parallel example?

Reply:
There is a way to write it inside a for comprehension like this:

```
for
    _ <- Future.unit
    f1 = Future { sleep(800); 1 }   // eventually returns 1
    f2 = Future { sleep(200); 2 }   // eventually returns 2
    f3 = Future { sleep(400); 3 }   // eventually returns 3
    r1 <- f1
    r2 <- f2
    r3 <- f3
  yield
      println(s"in the 'yield': ${delta()}")
      (r1 + r2 + r3)  
```    

I lik this way because it keeps the code blocks together, less change of code parts (1) and (2) to drift apart over time.
Even Viktor uses it. https://viktorklang.com/blog/Futures-in-Scala-protips-2.html :)

Reply:
I have added a commit that clarifies things further and adds a notice about computations run sequentially within a `for` expression.